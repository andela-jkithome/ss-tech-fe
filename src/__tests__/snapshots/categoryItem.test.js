import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import CategoryItem from '../../components/categoryItem';

configure({ adapter: new Adapter() });

const props = {
  category: 'science',
  cat: 'science',
  open: true,
  fetching: true,
  value: `Nothing can escape the gravity of a black hole, except for Chuck Norris. 
  Chuck Norris eats black holes. They taste like chicken.`,
  fetchfact: jest.fn(),
  factError: null,
};

it('Renders correctly when not open', () => {
  const component = renderer.create(
    <CategoryItem {...Object.assign({}, props, { open: false })} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when not fetching, is selected category and open', () => {
  const component = renderer.create(
    <CategoryItem {...Object.assign({}, props, { fetching: false })} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there was an error fetching fact', () => {
  const component = shallow(
    <CategoryItem
      {...Object.assign({}, props, {
        fetching: false,
        factError: 'Error Fetching Fact!',
      })}
    />,
  );
  expect(toJson(component)).toMatchSnapshot();
});

it('Renders correctly when fetching, is selected category and open', () => {
  const component = renderer.create(<CategoryItem {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when fetching and not selected category', () => {
  const component = renderer.create(
    <CategoryItem {...Object.assign({}, props, { cat: 'dev' })} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
