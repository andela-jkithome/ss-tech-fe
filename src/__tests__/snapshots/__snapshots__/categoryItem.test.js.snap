// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly when fetching and not selected category 1`] = `
<div
  className="ui grid container"
>
  <div
    className="column"
    columns={1}
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="ui blue center aligned segment"
          onClick={[Function]}
        >
          dev
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders correctly when fetching, is selected category and open 1`] = `
<div
  className="ui grid container"
>
  <div
    className="column"
    columns={1}
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="ui blue center aligned segment"
          onClick={[Function]}
        >
          science
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column jokemessage"
      >
        <div
          className="ui message"
        >
          <i
            aria-hidden="true"
            className="circle notched loading icon"
          />
          <div
            className="content"
          >
            <div
              className="header"
            >
              Just one second
            </div>
            We are fetching that joke for you.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders correctly when not fetching, is selected category and open 1`] = `
<div
  className="ui grid container"
>
  <div
    className="column"
    columns={1}
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="ui blue center aligned segment"
          onClick={[Function]}
        >
          science
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="ui message jokemessage"
        >
          <div
            className="content"
          >
            <div
              className="header"
            >
              Science Chuck Norris Fact
            </div>
            <p>
              Nothing can escape the gravity of a black hole, except for Chuck Norris. 
                Chuck Norris eats black holes. They taste like chicken.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders correctly when not open 1`] = `
<div
  className="ui grid container"
>
  <div
    className="column"
    columns={1}
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="ui blue center aligned segment"
          onClick={[Function]}
        >
          science
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders correctly when there was an error fetching fact 1`] = `
<Grid
  className="container"
>
  <GridColumn
    columns={1}
  >
    <GridRow>
      <GridColumn>
        <Segment
          color="blue"
          onClick={[Function]}
          textAlign="center"
        >
          science
        </Segment>
      </GridColumn>
    </GridRow>
    <ErrorDisplay
      header="Error Fetching Fact!"
      text="There was an error fetching the fact. Please try again"
    />
  </GridColumn>
</Grid>
`;
